byzcoin/bcadmin/clicontracts/README.md

```diff
@@ -61,4 +61,32 @@ Invoke an addProof on a deferred contract:
 ```bash
 # The --hash and --instID values are given when we spawn the deferred contract
 bcadmin contract deferred invoke addProof --hash ... --instID ... --iid 0
+```
+
+**Working scenario**:
+
+```bash
+# Run the nodes, create roster and set up the config
+./run_nodes.sh -n 5 -c -t -v 2
+export BC_CONFIG=~/GitHub/cothority/conode
+bcadmin create -roster ~/GitHub/cothority/conode/public.toml 
+
+# Add the rules specific to the value and deferred contracts.
+# We use the admin identity.
+bcadmin darc rule -rule spawn:value --identity ed25519:... 
+bcadmin darc rule -rule spawn:deferred --identity ed25519:...                                                         
+bcadmin darc rule -rule invoke:deferred.addProof --identity ed25519:...        
+bcadmin darc rule -rule invoke:deferred.execProposedTx --identity ed25519:...                                                                                                                                                     
+
+# Spawn a value contract, but redirect the transaction to the spawn of a deferred contract
+bcadmin contract value spawn --value myValue --redirect | bcadmin contract deferred spawn
+
+# Add the proof on the single instruction of the deferred transaction 
+# (the --hash and --instID values are given when we spawn the deferred contract)
+bcadmin contract deferred invoke addProof --hash ... --instID ... --iid 0
```

Comment:
I understand how you came up with this `bcadmin contract deferred invoke addProof` command, but it's just way too big. The cli tools are meant to be used by humans, and need to be short and possible to remember. For instance, `bcadmin -x ... | bcadmin deferred` to start one, and `bcadmin deferred sign $id` to sign one.

Reply:
Before anything, here is the convention I am using for the clicontrant commands:

```bash
   bcadmin contract CONTRACT { spawn  --bc <byzcoin config> 
                                      [--<arg name> <arg value>, ...]
                                      [--darc <darc id>] 
                                      [--sign <pub key>] 
                                      [--redirect],
                               invoke <command>
                                      --bc <byzcoin config>
                                      --instID <instance ID>
                                      [--<arg name> <arg value>, ...]
                                      [--darc <darc id>] 
                                      [--sign <pub key>],
                               get    --bc <byzcoin config>
                                      --instID <instance ID>,
                               delete --bc <byzcoin config>
                                      --instID <instance ID>
                                      [--darc <darc id>] 
                                      [--sign <pub key>]     
                             }
```


I knew I would need to fight for this case 😁, let me explain :

- **Short command line doesn't necessarily reduce the effort to remember**. I personally will have less of a tendency to forget if something is logical and follows a convention than if it's short. Having a convention also allows me to "rebuild" the command in case I forget it because it follows a standard, which brings the next point
- **Everyone one eventually forgets**, which means that, even if the command is shot, it is likely that you will have to look at the doc to re-learn the specific command you need to use. If a convention is used, one has only to remember the logic and convention used in order to fully appreciate the handling of clicontracts.
- **clicontract is a cli interface for a contract**, which is why I am not re-inventing nor simplifying commands, but rather **mapping** the contract implementation to a command line usage. This allows anyone that understands how contract are implemented to use the clicontract and vice-versa, without any cognitive effort requested to map special command to the corresponding contract function (in your proposal, you require users to understand that `bcadmin deferred` == `Spawn:deferred` and `bcadmin deferred sign` == `Invoke:deferred.addProof`. Establishing conventions and using them has the huge benefit of improving implementation, usage, maintenance, and learning of a system. In this case it also clearly describe what the system is doing.

Those points stand because there are more than one clicontracts. I would agree to use "special short" command it they were only one case of the deferred contract with - let's say - the value contract. But here I think it's very interesting to have a standard way to implement clicontracts, which is a very nice interface for handling them. 
(also I did not forget that somme of the command will move at the "contract" level, and --redirect will be renamed to --export).

Reply:
Maybe a way to satisfy the two conflicting points (easy to use vs following a logical standard) is to use aliases. Perhaps there's a `bcadmin deferred` that only contains aliases of the frequently used commands. Like `bcadmin deferred sign $inst --bc $config` = `bcadmin contract deferred invoke "sign" --bc $bc --instID $inst`. It could just be a shell script that uses `bcadmin` (not saying that's a good idea). Just like `git pull` is like an alias for fetching and then merging.

I think the current approach is fine. But when many more people start using deferred, we may need to provide an alias.

Reply:
I'm totally for it! Let's add an alias when/if the need arises.